### DISCUSSION TITLE
Two I2C devices and webserver (ESP32-CAM)

### DISCUSSION THREAD

**Discussion by lgouj** [arendst/Tasmota#15735] [Q&A] [answered]
Hello,
I'd like to use an ESP32-CAM with INA219 and ADS1115 devices. I've compiled a tasmota version using gitpod, enabling I2C support and the corresponding drivers (I've also enabled SH1106 and SR04 drivers for future use).
GPIO14 is used as SDA, GPIO15 as SCL. The bus adresses are different (default adresses). When the two devices are connected on the same bus, I can't connect to the webUI. If I use only one of them, everything is ok...

**Comment by sfromis** [comment]
Since gpio 14/15 are connected to the SPI for the camera module, they are not really free for other usage. Not sure of what outcome would be expected from conflicts there. 

If you want to see the preconfigured pin assignments, you can use "Configure Template" and select "ESP32-Cam" as "Based on". You do not need to save. Gpio 16,17 are red due to them being used for PSRAM.

**Comment by lgouj** [comment, marked as answer]
The problem may come from the 47k pullup resistor on GPIO15 that is present on that board. I've moved SCL and SDA wires on GPIO 1 and 3 (serial pins), added two 2.2k pullup resistors to 3V3, and it works now.

**Reply by sfromis** [reply]
Great that you got it to work. While I2C does need pullups, such resistors are already there on most breakout boards. I've never added pullups when connecting I2C sensors, but that is basically up to the mix of sensors I'm using.

**Reply by lgouj** [reply]
Using the same pins (GPIO 1 and 3), without the pullup resistors, the webserver doesn't work anymore. If i connect these two sensors to an arduino uno board, the datas are well transmitted through the I2C bus without the pullup resistors...It looks like a kind of conflict between the webserver and the bus...

**Reply by s-hadinger** [reply]
Webserver doesn't use any gpio. GPIO 1/3 are normally used for Serial UART. You may have a deeper conflict. As a general rule I would avoid using GPIO 1/3 for anything else than serial console
